src/pybind/mgr/telemetry/module.py

```diff
@@ -306,6 +306,43 @@ def gather_configs(self) -> Dict[str, List[str]]:
             'active_changed': sorted(list(active)),
         }
 
+    def get_heap_stats(self) -> Dict[str, dict]:
+        # Initialize result dict
+        result: Dict[str, dict] = defaultdict(lambda: defaultdict(int))
+
+        # Get list of osd ids from the metadata
+        osd_metadata = self.get('osd_metadata')
+
+        # Grab output from the "osd.x heap stats" command
+        for osd_id in osd_metadata:
+            cmd_dict = {
+                'prefix': 'heap',
+                'heapcmd': 'stats',
+                'id': str(osd_id),
+            }
+            r, outb, outs = self.osd_command(cmd_dict)
+            if r != 0:
+                self.log.debug("Invalid command dictionary.")
+                continue
+            else:
+                if "tcmalloc heap stats" in outs:
+                    values = [int(i) for i in outs.split() if i.isdigit()]
+                    categories = ['use_by_application', 'page_heap_freelist',
+                                  'central_cache_freelist', 'transfer_cache_freelist',
+                                  'thread_cache_freelists', 'malloc_metadata',
+                                  'actual_memory_used', 'released_to_os',
+                                  'virtual_address_space_used', 'spans_in_use',
+                                  'thread_heaps_in_use', 'tcmalloc_page_size']
+                    assert(len(values) == 12)
```

Comment:
in case there's some problem with the output it'd be more resilient to return nothing, instead of asserting, which would stop the telemetry report from getting sent